src/peep/staff.c

```diff
@@ -78,6 +79,49 @@ void game_command_update_staff_colour(int *eax, int *ebx, int *ecx, int *edx, in
 	*ebx = 0;
 }
 
+
+/*
+Select random skins for entertainers on palcement.
++ only select valid skins for the current scenery
+*/
+int selectRandomSkin(int staff_type)
+{
+	if (staff_type != 3)
+		return 0;
+
+	// Some code is 1:1 out of window_staff_options_mousedown in windows\staff.c, but is needed also here!
+
+	init_scenery();
+
+	int ebx = 0;
+	for (int i = 0; i < 19; i++) {
+		if (window_scenery_tab_entries[i][0] != -1) {
+			rct_scenery_set_entry* scenery_entry = g_scenerySetEntries[i];
+			ebx |= scenery_entry->var_10A;
+		}
+	}
+
+	uint16 no_entries = 0;
+	for (uint8 i = 0; i < 19; ++i) {
+		if (ebx & (1 << i)) {
+			no_entries++;
+		}
+	}
+	uint16 randVal = rand() % no_entries;
```

Comment:
Not hiring or placement or movement. But providing the game is in sync. all other actions should work. Regardless we should not deliberately not use synchronisation techniques just because it currently doesn't work properly.